### ISSUE TITLE
[editor] find all usages

### ISSUE THREAD

**Issue by dannypsnl** [racket-tw/sauron#157]
**Describe the solution you'd like**
When c+click a definition, there shall show a list of references to jump.

### With https://github.com/racket-tw/sauron/milestone/2

With the milestone, we have a local state for each file now. Now, each maintainer should broadcast for `(list 'using me)`-like messages to other maintainers, then the reply will build up to be a complete query result.

Based on this, each maintainer must first tell those maintainers are used by them, then they will ask back these maintainers the positions. There have two results
1. no using now, please remove me from your user list
2. still using, these are the positions(to report to the editor for jumping list)

In this method, maintaining the user list would be a problem, should it be a list of files or a list of thread-handle? Using thread-handle would be more efficient and easier to access a thread, but removing might be a problem(check if we have `thread-equal?`); files will not have this problem, but have to access the shared hashmap for thread-handle. Another problem is should we use list or hashmap to store? A list updating might be a trouble for big module like keyword module from Racket, let's see what would happen.

**Comment by dannypsnl** [comment]
Blocked by #166

**Comment by dannypsnl** [comment]
Another way is to let every file hold their reference to `file:id` as key, and ask every file when querying.

**Comment by dannypsnl** [comment]
```rkt
(define (make-record-maintainer file-path)
  (thread
   (thunk
    (define cached-record (collect-from file-path))
    (define depender (mutable-set))
    (for/async ([path (hash-keys (record-requires cached-record))])
      (thread-send (get-record-maintainer path #:wait? #t)
        (list 'depended (current-thread))))
    (let loop ()
      (match (thread-receive)
        [(list 'depended from)
         ; pseudo-code here
         (set-add! depender from)]
        ...)
      (loop)))))
```

**Comment by dannypsnl** [comment]
The next to try
- [x] replace `mutable-set` with `let loop` state then an immutable set?
    With the experiment below, we know the `list` takes the most memory, and `set` is the smallest, but no huge difference between `mutable-set`. Hence, I will postpone this to the future.

```racket
(define-syntax (test stx)
  (syntax-parse stx
    [(_ body ...)
     #'(begin
         (define b (current-memory-use))
         body ...
         (define a (current-memory-use))
         (- a b 400))]))

(test (list))
(test (mutable-set))
(test (set))
```

- [ ] limit only those maintainers in the project get `'depended` message and record them?
- [x] handle `'references` message of course
